src/rpcTest/scala/io/iohk/ethereum/rpcTest/RpcApiTests.scala

```diff
@@ -0,0 +1,1138 @@
+package io.iohk.ethereum.rpcTest
+
+import java.math.BigInteger
+import java.security.SecureRandom
+
+import akka.util.ByteString
+import io.iohk.ethereum.domain.Address
+import io.iohk.ethereum.jsonrpc.TransactionRequest
+import io.iohk.ethereum.keystore.{KeyStoreImpl, Wallet}
+import io.iohk.ethereum.rlp
+import io.iohk.ethereum.rpcTest.Tags.{MainNet, PrivNet, PrivNetNoMining}
+import io.iohk.ethereum.utils.{KeyStoreConfig, Logger}
+import org.bouncycastle.util.encoders.Hex
+import org.scalatest.{FlatSpec, Matchers}
+import org.web3j.protocol.admin.Admin
+import org.web3j.protocol.core.DefaultBlockParameter
+import org.web3j.protocol.core.methods.request.{EthFilter, Transaction}
+import org.web3j.protocol.core.methods.response.EthBlock.{TransactionHash, TransactionObject}
+import org.web3j.protocol.http.HttpService
+import io.iohk.ethereum.network.p2p.messages.CommonMessages.SignedTransactions.SignedTransactionEnc
+import org.web3j.protocol.core.methods.response.EthLog.{Hash, LogObject}
+import io.iohk.ethereum.rpcTest.TestContracts._
+import io.iohk.ethereum.rpcTest.TestData._
+import org.web3j.protocol.core.methods.response.EthLog
+
+import scala.collection.JavaConverters._
+import scala.language.implicitConversions
+
+class RpcApiTests extends FlatSpec with Matchers with Logger {
+
+  "Json rpc service" should "be listening before all tests" taggedAs(MainNet, PrivNet) in new ScenarioSetup {
+    val response = service.netListening().send()
+    response.isListening shouldBe true
+  }
+
+  it should "return correct protocol version" taggedAs(MainNet, PrivNet) in new ScenarioSetup {
+    val response = service.ethProtocolVersion().send()
+    hexToBigInt(response.getProtocolVersion) shouldEqual 63
+  }
+
+  it should "return correct client version" taggedAs(MainNet, PrivNet) in new ScenarioSetup {
+    val response = service.web3ClientVersion().send()
+    response.getWeb3ClientVersion shouldEqual "mantis/v0.1"
```

Comment:
Maybe this `mantis/v0.1` value should be in some config sort of type, because if version is changed this would fail